apriltags_ros/src/apriltag_detector.cpp

```diff
@@ -340,32 +339,45 @@ void AprilTagDetector::imageCb(const sensor_msgs::PointCloud2ConstPtr& cloud,
           transformedPose.getRotation().getY(),
           transformedPose.getRotation().getZ(),
           transformedPose.getRotation().getW());
-        tf::poseTFToMsg(transformedPose, pose);
+        tf::poseTFToMsg(transformedPose, april_tag_raw_pose);
         tf::poseTFToMsg(transformedPlanePose, planePose);
       }
 
       geometry_msgs::PoseStamped tag_pose;
-      tag_pose.pose = pose;
+      tag_pose.pose = april_tag_raw_pose;
       tag_pose.header = header;
 
+      AprilTagDetection tag_detection;
+      tag_detection.pose = tag_pose;
+      tag_detection.id = detection.id;
+      tag_detection.size = tag_size;
+      
```

Comment:
I think you can just grab all tags into the all_tag_detection_array here ```all_tag_detection_array.detections.push_back(tag_detection);```, just clear it at the start of every frame, and publish at end of frame processing. Here's the code from before we added validation: https://github.com/6RiverSystems/apriltags_ros/blob/old-indigo-devel/apriltags_ros/src/apriltag_detector.cpp#L100